Outdated jdependency dependency causes issues with other plugins using asm.

### Shadow Version

6.1.0

### Gradle Version

6.8.3

### Expected Behavior

Not cause errors due to conflicting asm versions with [Fabric loom](https://github.com/FabricMC/fabric-loom).

### Actual Behavior

Errors when trying to execute buildscript due to gradle choosing shadow's jdependency bundled asm instead of the asm pulled by [Fabric loom](https://github.com/FabricMC/fabric-loom).

### Gradle Build Script(s)

[Related buildscript](https://github.com/LambdAurora/LambdaControls/blob/ab8b78f75a213c95f98ed0e5e408f4e1078aada9/build.gradle)

Related error:
```
Caused by: java.lang.IllegalArgumentException
        at org.objectweb.asm.ClassVisitor.<init>(ClassVisitor.java:70)
        at net.fabricmc.tinyremapper.TinyRemapper$3.<init>(TinyRemapper.java:429)
        at net.fabricmc.tinyremapper.TinyRemapper.analyze(TinyRemapper.java:429)
        at net.fabricmc.tinyremapper.TinyRemapper.access$400(TinyRemapper.java:65)
        at net.fabricmc.tinyremapper.TinyRemapper$2.visitFile(TinyRemapper.java:411)
        at net.fabricmc.tinyremapper.TinyRemapper$2.visitFile(TinyRemapper.java:407)
        at net.fabricmc.tinyremapper.TinyRemapper.readFile(TinyRemapper.java:407)
        at net.fabricmc.tinyremapper.TinyRemapper.access$200(TinyRemapper.java:65)
        at net.fabricmc.tinyremapper.TinyRemapper$1$1.get(TinyRemapper.java:376)
        at net.fabricmc.tinyremapper.TinyRemapper$1$1.get(TinyRemapper.java:372)
```

### Content of Shadow JAR (`jar tf <jar file>` - post link to GIST if too long)

Unrelated to this issue as it doesn't even build.

### Additional information

This is due to the outdated `jdependency` used by the shadow plugin.
This is a shade issue in `jdependency` that has been solved in [jdependency 2.6.0](https://github.com/tcurdt/jdependency/commit/6e233125c48a0a977f9d443b154180b7341ecc2a).

--- a/build.gradle
+++ b/build.gradle
@@ -8,12 +8,12 @@ buildscript {
         }
     }
     dependencies {
-        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.8.1'
-        classpath "org.jfrog.buildinfo:build-info-extractor-gradle:4.9.9"
-        classpath "com.gradle.publish:plugin-publish-plugin:0.11.0"
-        classpath 'org.ajoberstar:grgit:2.1.1'
-        classpath 'org.ajoberstar:gradle-git-publish:0.3.3'
-        classpath "com.moowork.gradle:gradle-node-plugin:1.2.0"
+        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.8.5'
+        classpath "org.jfrog.buildinfo:build-info-extractor-gradle:4.21.0"
+        classpath "com.gradle.publish:plugin-publish-plugin:0.14.0"
+        classpath 'org.ajoberstar:grgit:4.1.0'
+        classpath 'org.ajoberstar:gradle-git-publish:3.0.0'
+        classpath "com.moowork.gradle:gradle-node-plugin:1.3.1"
     }
 }
 

--- a/build.gradle
+++ b/build.gradle
@@ -11,7 +11,7 @@ buildscript {
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.8.5'
         classpath "org.jfrog.buildinfo:build-info-extractor-gradle:4.21.0"
         classpath "com.gradle.publish:plugin-publish-plugin:0.14.0"
-        classpath 'org.ajoberstar:grgit:4.1.0'
+        classpath 'org.ajoberstar:grgit:2.3.0'
         classpath 'org.ajoberstar:gradle-git-publish:3.0.0'
         classpath "com.moowork.gradle:gradle-node-plugin:1.3.1"
     }

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -18,7 +18,7 @@ dependencies {
     testImplementation("org.spockframework:spock-core:1.3-groovy-2.5") {
         exclude module: 'groovy-all'
     }
-    testImplementation 'xmlunit:xmlunit:2.8.2'
+    testImplementation 'xmlunit:xmlunit:2.8.0'
     testImplementation 'org.apache.commons:commons-lang3:3.12.0'
     testImplementation 'com.google.guava:guava:30.1-jre'
 }

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -18,7 +18,7 @@ dependencies {
     testImplementation("org.spockframework:spock-core:1.3-groovy-2.5") {
         exclude module: 'groovy-all'
     }
-    testImplementation 'xmlunit:xmlunit:2.8.0'
+    testImplementation 'xmlunit:xmlunit:1.3'
     testImplementation 'org.apache.commons:commons-lang3:3.12.0'
     testImplementation 'com.google.guava:guava:30.1-jre'
 }

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -20,5 +20,5 @@ dependencies {
     }
     testImplementation 'xmlunit:xmlunit:1.6'
     testImplementation 'org.apache.commons:commons-lang3:3.12.0'
-    testImplementation 'com.google.guava:guava:30.1-jre'
+    testImplementation 'com.google.guava:guava:30.1.1-jre'
 }

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -2,23 +2,23 @@ dependencies {
 
     shadow localGroovy()
     shadow gradleApi()
-    shadow 'org.codehaus.groovy:groovy-backports-compat23:2.4.15'
+    shadow 'org.codehaus.groovy:groovy-backports-compat23:3.0.7'
 
     implementation 'org.jdom:jdom2:2.0.6'
-    implementation 'org.ow2.asm:asm:9.0'
-    implementation 'org.ow2.asm:asm-commons:9.0'
-    implementation 'commons-io:commons-io:2.5'
-    implementation 'org.apache.ant:ant:1.9.7'
-    implementation 'org.codehaus.plexus:plexus-utils:3.0.24'
-    implementation "org.apache.logging.log4j:log4j-core:2.13.3"
-    implementation('org.vafer:jdependency:2.1.1') {
+    implementation 'org.ow2.asm:asm:9.1'
+    implementation 'org.ow2.asm:asm-commons:9.1'
+    implementation 'commons-io:commons-io:2.8.0'
+    implementation 'org.apache.ant:ant:1.10.9'
+    implementation 'org.codehaus.plexus:plexus-utils:3.3.0'
+    implementation "org.apache.logging.log4j:log4j-core:2.14.1"
+    implementation('org.vafer:jdependency:2.6.0') {
         exclude group: 'org.ow2.asm'
     }
 
-    testImplementation("org.spockframework:spock-core:1.0-groovy-2.4") {
+    testImplementation("org.spockframework:spock-core:1.3-groovy-2.5") {
         exclude module: 'groovy-all'
     }
-    testImplementation 'xmlunit:xmlunit:1.3'
-    testImplementation 'commons-lang:commons-lang:2.6'
-    testImplementation 'com.google.guava:guava:17.0'
+    testImplementation 'xmlunit:xmlunit:2.8.2'
+    testImplementation 'org.apache.commons:commons-lang3:3.12.0'
+    testImplementation 'com.google.guava:guava:30.1-jre'
 }

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -18,7 +18,7 @@ dependencies {
     testImplementation("org.spockframework:spock-core:1.3-groovy-2.5") {
         exclude module: 'groovy-all'
     }
-    testImplementation 'xmlunit:xmlunit:1.3'
+    testImplementation 'xmlunit:xmlunit:1.6'
     testImplementation 'org.apache.commons:commons-lang3:3.12.0'
     testImplementation 'com.google.guava:guava:30.1-jre'
 }

--- a/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/util/file/TestFileHelper.groovy
+++ b/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/util/file/TestFileHelper.groovy
@@ -1,7 +1,7 @@
 package com.github.jengelman.gradle.plugins.shadow.util.file;
 
 import org.apache.commons.io.FileUtils
-import org.apache.commons.lang.StringUtils
+import org.apache.commons.lang3.StringUtils
 import org.apache.tools.ant.Project
 import org.apache.tools.ant.taskdefs.Expand
 import org.apache.tools.ant.taskdefs.Tar

--- a/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/util/file/TestNameTestDirectoryProvider.java
+++ b/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/util/file/TestNameTestDirectoryProvider.java
@@ -1,7 +1,7 @@
 package com.github.jengelman.gradle.plugins.shadow.util.file;
 
 
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.junit.rules.MethodRule;
 import org.junit.rules.TestRule;
 import org.junit.runner.Description;

--- a/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/util/repo/maven/MavenScope.groovy
+++ b/src/test/groovy/com/github/jengelman/gradle/plugins/shadow/util/repo/maven/MavenScope.groovy
@@ -1,6 +1,6 @@
 package com.github.jengelman.gradle.plugins.shadow.util.repo.maven
 
-import org.apache.commons.lang.StringUtils
+import org.apache.commons.lang3.StringUtils
 
 class MavenScope {
     Map<String, MavenDependency> dependencies = [:]

